cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperator.java

```diff
@@ -0,0 +1,804 @@
+                                                                RebalanceOptions.RebalanceOptionsBuilder rebalanceOptionsBuilder) {
+        switch (rebalanceAnnotation) {
+            case none:
+                log.debug("{}: No {} annotation set", reconciliation, ANNO_STRIMZI_IO_REBALANCE);
+                return Future.succeededFuture(clusterRebalance.getStatus());
+            case approve:
+                return requestRebalance(reconciliation, host, apiClient, false, rebalanceOptionsBuilder);
+            case refresh:
+                return requestRebalance(reconciliation, host, apiClient, true, rebalanceOptionsBuilder);
+            default:
+                log.warn("{}: Ignore annotation {}={}", reconciliation, ANNO_STRIMZI_IO_REBALANCE, rebalanceAnnotation);
+                return Future.succeededFuture(clusterRebalance.getStatus());
+        }
+    }
+
+    /**
+     * This method handles the transition from {@code Rebalancing} state.
+     * It starts a periodic timer in order to check the status of the ongoing rebalance processing on Cruise Control side.
+     * In order to do that, it calls the related Cruise Control REST API about asking the user task status.
+     * When the rebalance is finished, the next state is {@code Ready}.
+     * If the user sets the strimzi.io/rebalance=stop annotation, it calls the Cruise Control REST API for stopping the ongoing task
+     * and then move to the {@code Stopped} state.
+     * If the user sets any other values for the strimzi.io/rebalance, it is just ignored and the user task checks just continues.
+     * This method holds the lock until the rebalance is finished, the ongoing task is stopped or any exception is raised.
+     *
+     * @param reconciliation Reconciliation information
+     * @param host Cruise Control service to which sending the REST API requests
+     * @param apiClient Cruise Control REST API client instance
+     * @param clusterRebalance Current {@code KafkaClusterRebalance} resource
+     * @param rebalanceAnnotation The current value for the strimzi.io/rebalance annotation
+     * @return a Future with the next {@code KafkaClusterRebalanceStatus} bringing the state
+     */
+    private Future<KafkaClusterRebalanceStatus> onRebalancing(Reconciliation reconciliation,
+                                                              String host, CruiseControlApi apiClient,
+                                                              KafkaClusterRebalance clusterRebalance,
+                                                              RebalanceAnnotation rebalanceAnnotation) {
+        Promise<KafkaClusterRebalanceStatus> p = Promise.promise();
+        if (rebalanceAnnotation == RebalanceAnnotation.none) {
+            log.info("{}: Arming Cruise Control rebalance user task status timer", reconciliation);
+            String sessionId = clusterRebalance.getStatus().getSessionId();
+            AtomicInteger ccAPIErrorCount = new AtomicInteger();
```

Comment:
`ccApiErrorCount` would be more consistent with most of the rest of the project, I think. (Or have I lost this battle?)